### PR TITLE
Changed StatusBar base class, re-organized level loading, implemented…

### PR DESCRIPTION
**Pull request by PaxtonWong**
… PlayerStates for tracking health/stamina/statuses. Implemented death after Player Health is depleted.

### CODE REVIEW

File: Assets/Health.cs [hunk 1]
```diff
@@ -3,14 +3,33 @@
...
+
     public Health()
     {
 
     }
 
-    public Health(int maxHealth) : base(maxHealth)
+    public Health(int maxHealth, PlayerStatus playrStatus) : base(maxHealth, playrStatus)
+    {
+        playerStatus = playrStatus;
```
Comment (cnavasca95) [inline Assets/Health.cs]: playerStatus?

File: Assets/Stamina.cs [hunk 2]
```diff
@@ -3,15 +3,16 @@
...
     public Stamina()
     {
 
     }
 
-    public Stamina(int maxStamina) : base (maxStamina)
+    public Stamina(int maxStamina, PlayerStatus playrStatus) : base (maxStamina, playrStatus)
     {
-
+        playerStatus = playrStatus;
```
Comment (cnavasca95) [inline Assets/Stamina.cs]: playerStatus?

File: Assets/StatusBar.cs [hunk 3]
```diff
@@ -3,42 +3,40 @@
...
 
     }
     // Use this for initialization
-    public StatusBar(int maxed)
+    public StatusBar(int maxed, PlayerStatus playrStatus)
     {
         max = maxed;
         current = max;
-	}
+        playerStatus = playrStatus;
```
Comment (cnavasca95) [inline Assets/StatusBar.cs]: playerStatus?

File: Assets/StatusBar.cs [hunk 4]
```diff
@@ -70,4 +68,28 @@ public float percent()
         return (float)current / max;
     }
 
+    public void setRecoveryRate(int recoveryAmount, float timeDelta)
+    {
+        recoveryRate = recoveryAmount;
+        timeBetweenRecoveries = timeDelta;
```
Comment (cnavasca95) [inline Assets/StatusBar.cs]: timeDelta is pretty vague. At least name it so the user knows the time unit

File: Assets/StatusBar.cs [hunk 5]
```diff
@@ -70,4 +68,28 @@ public float percent()
...
+    }
+
+    public bool isRecovering()
+    {
+        return recoveryRate > 0 && timeBetweenRecoveries >= 0.1f && getRemaining() < getMax();
```
Comment (cnavasca95) [inline Assets/StatusBar.cs]: what is the point of the .1f check?

File: Assets/StatusBar.cs [hunk 6]
```diff
@@ -3,42 +3,40 @@
 
 public class StatusBar : MonoBehaviour {
 
-    private int current;
-    private int max;
+    public PlayerStatus playerStatus;
+    public int current = 100;
+    private int max = 100;
```
Comment (cnavasca95) [inline Assets/StatusBar.cs]: Match these visibilities?

**Review by cnavasca95 — COMMENTED** [review]
Mostly a few small things.

File: Assets/Health.cs [hunk 7]
```diff
@@ -9,9 +9,9 @@ public Health()
 
     }
 
-    public Health(int maxHealth, PlayerStatus playrStatus) : base(maxHealth, playrStatus)
+    public Health(int maxHealth, PlayerStatus playerStatus) : base(maxHealth, playerStatus)
     {
-        playerStatus = playrStatus;
+        this.playerStatus = playerStatus;
```
Comment (kgwong) [inline Assets/Health.cs]: do you need this if base does this?

File: Assets/Stamina.cs [hunk 8]
```diff
@@ -10,9 +10,9 @@ public Stamina()
 
     }
 
-    public Stamina(int maxStamina, PlayerStatus playrStatus) : base (maxStamina, playrStatus)
+    public Stamina(int maxStamina, PlayerStatus playerStatus) : base (maxStamina, playerStatus)
     {
-        playerStatus = playrStatus;
+        this.playerStatus = playerStatus;
```
Comment (kgwong) [inline Assets/Stamina.cs]: same as above

File: Assets/PlayerStatus.cs [hunk 9]
```diff
@@ -15,7 +15,7 @@ void Start ()
         playerStamina = new Stamina(100, this);
 
         //set these values as you like, make sure to keep second parameter above 0.1f or will not update
-        playerHealth.setRecoveryRate(1, 0.25f);
+        playerHealth.setRecoveryRate(1, 0.25f); //(recovery amount, amount of time between recovery)
```
Comment (kgwong) [inline Assets/PlayerStatus.cs]: remove comment

File: Assets/PlayerStatus.cs [hunk 10]
```diff
@@ -15,7 +15,7 @@ void Start ()
         playerStamina = new Stamina(100, this);
 
         //set these values as you like, make sure to keep second parameter above 0.1f or will not update
```
Comment (kgwong) [inline Assets/PlayerStatus.cs]: add an assert inside of setRecoveryRate that checks this condition

**Review by kgwong — APPROVED** [review]

File: Assets/StatusBar.cs [hunk 11]
```diff
@@ -5,10 +5,12 @@ public class StatusBar : MonoBehaviour {
 
     public PlayerStatus playerStatus;
     protected int current = 100;
-    private int max = 100;
+    protected int max = 100;
 
-    int recoveryRate = 0;
-    float timeBetweenRecoveries = 0.0f;
+    
+    int recoveryAmount = 0;
```
Comment (kgwong) [inline Assets/StatusBar.cs]: add scope modifiers (public/private/protected)

**Review by kgwong — COMMENTED** [review]
if previous comments conflict with what's in this commit, then use your own discretion on what you should push

**Pull request closed without merging**